server/app/models/host_node.rb

```diff
@@ -42,6 +42,10 @@ class Error < StandardError
 
   scope :connected, -> { where(connected: true) }
 
+  after_destroy do |node|
+    node.containers.unscoped.delete_all
```

Comment:
Currently there's no callback on the container model, so it is safe. But to be more future-proof, I'll change it to use destroy so that we don't have to worry when adding possible callbacks on the container model.